domain/modelmanager/service/service_test.go

```diff
@@ -0,0 +1,1580 @@
+// [coreerrors.NotValid].
+func (s *serviceSuite) TestRemoveNonActivateModelNotValid(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	svc := NewService(
+		s.mockModelRemover,
+		s.mockState,
+		loggertesting.WrapCheckLog(c),
+	)
+
+	err := svc.RemoveNonActivatedModel(c.Context(), coremodel.UUID(""))
+	c.Check(err, tc.ErrorIs, coreerrors.NotValid)
+}
+
+// TestRemoveNonActivatedModelNotFound tests that if a request is made to remove
+// a non active model that doesn't exist the caller gets back an error
+// satisfying [modelerrors.NotFound].
+func (s *serviceSuite) TestRemoveNonActivatedModelNotFound(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	svc := NewService(
+		s.mockModelRemover,
+		s.mockState,
+		loggertesting.WrapCheckLog(c),
+	)
+	modelUUID := modeltesting.GenModelUUID(c)
+
+	s.mockState.EXPECT().RemoveNonActivatedModel(gomock.Any(), modelUUID).Return(
+		modelerrors.NotFound,
+	)
+	err := svc.RemoveNonActivatedModel(c.Context(), modelUUID)
+	c.Check(err, tc.ErrorIs, modelerrors.NotFound)
+}
+
+// TestRemoveAlreadyActivatedModel tests that if a caller asks for a non
+// activated model to be removed and the model has already been active the
+// caller gets back an error satisfying [modelmanagererrors.AlreadyActivated].
+func (s *serviceSuite) TestRemoveAlreadyActivatedModel(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	svc := NewService(
+		s.mockModelRemover,
+		s.mockState,
+		loggertesting.WrapCheckLog(c),
+	)
+	modelUUID := modeltesting.GenModelUUID(c)
+
+	s.mockState.EXPECT().RemoveNonActivatedModel(gomock.Any(), modelUUID).Return(
+		modelmanagererrors.AlreadyActivated,
+	)
+	err := svc.RemoveNonActivatedModel(c.Context(), modelUUID)
+	c.Check(err, tc.ErrorIs, modelmanagererrors.AlreadyActivated)
+}
+
+// TestRemoveNonActivatedModel is testing removing a non activated model with
+// the option of having the model db deleted as part of the operation.
+func (s *serviceSuite) TestRemoveNonActivatedModelWithDeleteDB(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	svc := NewService(
+		s.mockModelRemover,
+		s.mockState,
+		loggertesting.WrapCheckLog(c),
+	)
+	modelUUID := modeltesting.GenModelUUID(c)
+
+	s.mockState.EXPECT().RemoveNonActivatedModel(gomock.Any(), modelUUID).Return(
+		nil,
+	)
+	s.mockModelRemover.EXPECT().DeleteDB(modelUUID).Return(nil)
+	err := svc.RemoveNonActivatedModel(c.Context(), modelUUID)
+	c.Check(err, tc.ErrorIsNil)
+}
+
+// TestRemoveNonActivatedModelWithNoDeleteDB is testing removing a non activated
+// model and that the model db is not deleted as part of the operation.
+func (s *serviceSuite) TestRemoveNonActivatedModelWithNoDeleteDB(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	svc := NewService(
+		s.mockModelRemover,
+		s.mockState,
+		loggertesting.WrapCheckLog(c),
+	)
+	modelUUID := modeltesting.GenModelUUID(c)
+
+	s.mockState.EXPECT().RemoveNonActivatedModel(gomock.Any(), modelUUID).Return(
+		nil,
+	)
+	s.mockModelRemover.EXPECT().DeleteDB(modelUUID).Return(nil)
```

Comment:
This is the incorrect bit of the test, right? We should not be calling this if delete db is false.